### PR TITLE
Fix POETRY_REPOSITORIES_ variable docs

### PR DESCRIPTION
**Pull request by SamambaMan**
POETRY_REPOSITORIES_<NAME>_URL wrong in the documentation

# Pull Request Check List

Resolves: #8491

No changes are required to the code.

### CODE REVIEW

**Comment by dimbleby** [comment]
it's not so simple as that: `POETRY_REPOSITORIES_<NAME>_URL` is specifically for setting the url associated with a repository ie it corresponds to `repositories.<name>.url`.

I can believe that there are clarity improvements to be made here and it would be good to see that: but I don't think this MR as it stands helps with that

**Comment by SamambaMan** [comment]
What do you suggest?
I merely wrote the same as it in the code (i think) for doing as the manual won't work.

Do you want me to explain that the <name> should match naming in sources?

**Comment by dimbleby** [comment]
the current doc has a heading `repositories.name`.  You are reporting - I think rightly - that as an environment variable this isn't very useful because you never want to configure only a repository name: you want to configure sub-properties of the repository (in particular, the URL).

However your MR leaves that heading alone, so now the doc is not self-consistent.

I realise that disentangling this is more than the four character fix you were hoping for!

**Comment by SamambaMan** [comment]
Oh, i get it now.
This should cover other parameters as priority as well, right?
I'm hovering the code and this is not supported code wise yet also, right?

**Comment by dimbleby** [comment]
As I understand it, global repository configuration is only interesting during publishing, so eg priority is irrelevant - I think url is the only interesting property.

**Comment by SamambaMan** [comment]
So...? :D

**Comment by dimbleby** [comment]
So: put some or more of that in the docs!

**Comment by SamambaMan** [comment]
@dimbleby check if this is better.
Also included fixes for basic auth as tested in https://github.com/python-poetry/poetry/blob/47a3a19c442c21aa5b2836ea092dda8ae4d13151/tests/utils/test_authenticator.py#L355

**Comment by SamambaMan** [comment]
Can I improve this PR in any way?
Let me know

File: docs/configuration.md [hunk 1]
```diff
@@ -437,19 +437,21 @@ If set to `false`, Python version used during Poetry installation is used.
...
-### `repositories.<name>`
+### `repositories.<name>.url`
 
 **Type**: `string`
 
-**Environment Variable**: `POETRY_REPOSITORIES_<NAME>`
+**Environment Variable**: `POETRY_REPOSITORIES_<NAME>_URL`
 
-Set a new alternative repository. See [Repositories]({{< relref "repositories" >}}) for more information.
+Set a new alternative repository URL for `name` repository
```
Comment (radoering) [inline docs/configuration.md:446]:
```suggestion
Set the repository URL for `<name>`.
```

File: docs/configuration.md [hunk 2]
```diff
@@ -437,19 +437,21 @@ If set to `false`, Python version used during Poetry installation is used.
...
 
-### `http-basic.<name>`:
+See [Repositories]({{< relref "repositories" >}}) for more information.
```
Comment (radoering) [inline docs/configuration.md:448]:
```suggestion
See [Publishable Repositories]({{< relref "repositories#publishable-repositories" >}}) for more information.
```

**Review by radoering — CHANGES_REQUESTED** [review]

**Review by radoering — APPROVED** [review]

**Pull request merged**